crates/swc_atoms/src/lib.rs

```diff
@@ -44,12 +44,37 @@ include!(concat!(env!("OUT_DIR"), "/js_word.rs"));
 ///   "longer than xx" as this is a type.
 /// - Raw values.
 #[derive(Clone)]
-pub struct Atom(ThinArc<HeaderWithLength<()>, u8>);
+pub struct Atom(Inner);
 
-fn _assert_size() {
-    let _static_assert_size_eq = std::mem::transmute::<Atom, usize>;
+#[derive(Clone)]
+enum Inner {
+    Arc(ThinArc<HeaderWithLength<()>, u8>),
+    Static(&'static str),
 }
 
+impl Inner {
+    fn new<S>(s: S) -> Self
+    where
+        Arc<str>: From<S>,
+        S: AsRef<str>,
+    {
+        let len = s.as_ref().as_bytes().len();
+
+        Self::Arc(ThinArc::from_header_and_slice(
+            HeaderWithLength::new((), len),
+            s.as_ref().as_bytes(),
+        ))
+    }
+
+    fn new_static(s: &'static str) -> Self {
+        Self::Static(s)
+    }
+}
+
+// fn _assert_size() {
+//     let _static_assert_size_eq = std::mem::transmute::<Atom, usize>;
```

Comment:
@kdy1 The `transmute` seems not to work, after making `Atom(ThinArc<HeaderWithLength<()>, u8>)` to `Atom(Inner)`.

Is this expected?
![image](https://user-images.githubusercontent.com/49502170/212448481-9b7b6fd7-acdc-46d5-9d97-1e8e817b8dc0.png)

Reply:
Yes, but it will degrade performance greatly.

I didn't implement static support because it's hard

Reply:
I get the gist. We need something like `ThinStaticStr`. It's indeed hard after looking into [thin_tr](https://github.com/thomcc/thin_str). I will try it, but it will take a while.

If anyone wants to try this, feel free to do it.